src/windows/editor_object_selection.c

```diff
@@ -944,8 +962,22 @@ static void window_editor_object_selection_dropdown(rct_window *w, int widgetInd
 
 	switch (widgetIndex) {
 	case WIDX_FILTER_DROPDOWN:
-		_filter_flags ^= (1 << dropdownIndex);
-		gConfigInterface.object_selection_filter_flags = _filter_flags;
+		if (dropdownIndex == 5) {
+			_filter_selected = !_filter_selected;
+			if (_filter_selected && _filter_nonselected) {
```

Comment:
I'm not sure if I am understanding you correctly. What you're saying to do sounds like it would set non-select to false, even if someone was checking non-select.